addons/survey/tests/test_survey_results.py

```diff
@@ -0,0 +1,279 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.addons.survey.controllers.main import Survey
+from odoo.addons.survey.tests import common
+from odoo.osv import expression
+from odoo import fields
+
+
+class TestSurveyResults(common.TestSurveyCommon):
+
+    def setUp(self):
+        self.certification_survey_id = self.ref('survey.vendor_certification')
+        self.quiz_survey_id = self.ref('survey.survey_demo_quiz')
+        self.UserInput = self.env['survey.user_input']
+        self.UserInputLine = self.env['survey.user_input.line']
+        self.QuestionAnswer = self.env['survey.question.answer']
+
+    def _check_filters_results(self, post, survey_id, lines_domain):
```

Comment:
I would specify the "performance" part in the method name as well. like "_results_and_query_count"

Reply:
Oh and you should pass a specific message from each case, so that if one assert fails here, we know which case it was.
```suggestion
    def _check_filters_results(self, post, survey_id, lines_domain, msg):
        ... 
        self.assertEqual(X, 1, msg)
```